shell/platform/android/test/io/flutter/plugin/editing/InputConnectionAdaptorTest.java

```diff
@@ -318,6 +319,13 @@ public void testSendKeyEvent_delKeyDeletesBackward() {
     int selStart = 29;
     Editable editable = sampleRtlEditable(selStart, selStart);
     InputConnectionAdaptor adaptor = sampleInputConnectionAdaptor(editable);
+    adaptor.setGetOffsetBefore(
```

Comment:
This seems to just adjust an old test, where is the test that exercises the new logic?

Reply:
This is a reimplementation of `InputConnectionAdaptor` delete key handling that uses a JNI function instead of an Android system API to find the character boundary.  Unfortunately the new part is JNI native code, and I don't think the embedding has infrastructure for testing that.  The Robolectric tests do not load the engine native code library.

Reply:
I've looked into this in the past, let me try to remember.  There are 2 ways you can make unit tests, robolectric or a slower test that runs on a simulator.  Robolectric can load JNI but your JNI library has to be compiled for the desktop operating system, which might be easy or near impossible depending on what you are doing.  It's near impossible you are calling native android libraries.